### PR TITLE
CXF-98497: Fix failing GHA Docs

### PR DESCRIPTION
**Pull request by d-bhola**
1. Modified GHA jobs by adding scripts for running terraform-docs. My changes include four jobs-
   - Install terraform-docs
   - Render Terraform docs inside the README.md files of examples 
   - Render Terraform docs inside the README.md files of modules
   - Replace EXAMPLE_PATH placeholders inside the README.md files of examples

2. Minor changes to examples-terraform-docs.yaml config file

### CODE REVIEW

**Comment by d-bhola** [comment]
Tested these changes. GHA jobs are passing [here](https://github.com/equinix/terraform-equinix-fabric/actions/runs/10322345359/workflow)

File: .github/workflows/documentation.yaml [hunk 1]
```diff
@@ -21,36 +21,31 @@ jobs:
...
-          config-file: templates/examples-terraform-docs.yaml
-          recursive: true
-          recursive-path: examples
-          output-file: README.md
-          args: --sort-by required
-          indention: 2
-          git-push: "false"
+        run: |
+          find examples -type d -mindepth 1 -maxdepth 1 | while read dir; do
+            terraform-docs -c "./templates/examples-terraform-docs.yaml" --output-file "README.md" "$dir"
```
Comment (d-bhola) [inline .github/workflows/documentation.yaml:34]: @thogarty I couldn't find a corresponding attribute (in GHA) to support the last parameter in terraform docs command( "$dir" ) which defines inside which directory the terraform-docs command would run. Therefore, I decided to use this approach of using running a script.

Comment (thogarty) [reply .github/workflows/documentation.yaml:34]: I support the approach. Still leverages terraform-docs just not directly with their GHA job. 

I was wondering why a pull request wasn't created at the the end of the action though? https://github.com/equinix/terraform-equinix-fabric/actions/runs/10322345359/job/28577358771

In the action you ran it should create a PR with the updates that it made during the earlier jobs.

**Review by thogarty — APPROVED** [review]
LGTM, thank you @d-bhola !

**Pull request merged**